src/helper/service.hpp

```diff
@@ -68,10 +70,21 @@ class service {
         return service_config_;
     }
 
+    std::optional<sampler::scope> schema_extraction_sampled()
+    {
+        if (!schema_extraction_enabled_) {
+            return std::nullopt;
+        }
+
+        return schema_sampler_.get();
+    }
+
 protected:
     std::shared_ptr<engine> engine_{};
     std::shared_ptr<service_config> service_config_{};
     dds::remote_config::client_handler::ptr client_handler_{};
+    sampler schema_sampler_;
+    bool schema_extraction_enabled_;
```

Comment:
We can get rid of this boolean flag altogether and just set the sampling to 0 when set to false in the configuration.

Reply:
Done